### ISSUE TITLE
SSH tunnel crashes app every time 

### ISSUE THREAD

**Issue by leehblue** [Alecaddd/sequeler#259]
When I try to connect to MySQL through SSH Tunnel the app crashes every time on Elementary OS. This happens when providing a password and when using SSH keys.

**Comment by Alecaddd** [comment]
Uh, that's not good, sorry for the issue.
I need more info in order to find the problem.
Did you install Sequeler from the app center?
Can you connect to your mysql via ssh from the terminal?
Can you launch sequeler from the terminal in debug mode and paste here the log?
`G_MESSAGES_DEBUG=all com.github.alecaddd.sequeler`

Thanks for using Sequeler :+1:

**Comment by leehblue** [comment]
Yes. I installed it through the App Center. Yes I can connect to the remote
MySQL server through the command line. I can also connect using MySQL
Workbench through an SSH tunnel.

In Sequeler, what happens is I click the connect button and it looks like
it tries to connect but about 1 second after clicking the connect button
the whole app crashes.

I am using the IP Address if the remote server not the domain.

I can successfully connect to local MySQL databases and using the App is
very nice. Definitely worth the price. Thank you so much for putting your
time into creating this app! It so much nicer than using MySQL Workbench
for the type of work I’m doing - mostly custom development for WordPress.

**Comment by Alecaddd** [comment]
Thanks for the info.
I too use the IP address in the SSH host, and that never gave me any issue.
I'll investigate and see if I can reproduce the error.

Whenever you have time, having the crash log from the terminal would be super useful.
Cheers

**Comment by leehblue** [comment]
Here's what happens when I run Sequeler from the command line and try to
open a remote database through an SSH tunnel.

$ G_MESSAGES_DEBUG=all com.github.alecaddd.sequeler
** (Sequeler:31362): DEBUG: 23:26:33.338: Settings.vala:163: Loading
settings from schema 'com.github.alecaddd.sequeler'
** (Sequeler:31362): DEBUG: 23:26:33.339: UpgradeManager.vala:30: Current
Version
(Sequeler:31362): Gtk-DEBUG: 23:26:33.374: Connecting to session manager
** (Sequeler:31362): DEBUG: 23:26:38.324: PasswordManager.vala:49: Unable
to fetch password in libsecret keyring for
com.github.alecaddd.sequeler.29999
** (Sequeler:31362): DEBUG: 23:26:38.324: ConnectionManager.vala:154:
Opening tunnel 0x55f2c968e4a0
** (Sequeler:31362): DEBUG: 23:26:38.654: ConnectionManager.vala:218:
Authentication methods: publickey,password
Segmentation fault

**Comment by leehblue** [comment]
I just realized something. I'm running PHP FPM which is listening in port
9000 and I think that's what's causing the conflict.

I switched my PHP FPM port to something else and now I'm able to connect to
remote databases!

This is looking great!

**Comment by Alecaddd** [comment]
Oh wow, that's interesting.
I'm using Valet Linux on elementary and I don't have any conflict with PHP.
Let's see if I can figure out a way to prevent the segfault crash if the port used is busy.

**Comment by leehblue** [comment]
I do a lot of work with WordPress and need to test and support multiple
versions of PHP so I use PHP Brew to install a bunch of PHP versions then I
proxy from Apache to whichever version of PHP I need to use. It's not the
normal PHP setup, but I do know that the default port for PHP FPM to listen
to tends to be 9000 which is the same port Sequeler likes to use. So, I
just changed my PHP FPM to listen to 9001.

**Comment by jameron** [comment]
Does the ssh tunnel work with AWS .pem keys?

**Comment by Salmondx** [comment]
Have the exact same problem with sequeler installed via flatpak.
Crashes during connection/test connection, on step when `Trying to connect to localhost:9000`. On my installation this port is not occupied so its even more weird.

**Comment by Alecaddd** [comment]
Is this issue still present with the latest release? I did some improvements to the SSH tunnel.

**Comment by gugacavalieri** [comment]
@jameron 

Has anyone got it to work with AWS SSH Tunnel and private `.pem` keys?

**Comment by fbnlsr** [comment]
Trying to connect to my Laravel Homestead database through SSH, and Sequeler crashes when testing connection. I briefly see "Opening SSH tunnel" and the app crashes and closes. I'm using Pop!_OS 20.04

**Comment by titojankowski** [comment]
Following up here, anyone able to login to an AWS MySQL db using the private .pem key using this app?
